src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -6338,36 +6339,124 @@ subroutine atm_rk_dynamics_substep_finish( state, diag, dynamics_substep, dynami
+               rho_zz_1(k,iCell) = rho_zz_2(k,iCell)
+            end do
+         end do
 
+         !$acc loop gang worker
+         do iCell = cellStart,cellEnd
+            !$acc loop vector
+            do k = 1,nVertLevels+1
+               rw_save(k,iCell) = rw(k,iCell)
+               w_1(k,iCell) = w_2(k,iCell)
+            end do
+         end do
+         !$acc end parallel
       end if
 
+
       if (dynamics_substep == 1) then
-         ruAvg_split(:,edgeStart:edgeEnd) = ruAvg(:,edgeStart:edgeEnd)
-         wwAvg_split(:,cellStart:cellEnd) = wwAvg(:,cellStart:cellEnd)
+         !$acc parallel
+         !$acc loop gang worker
+         do iEdge = edgeStart,edgeEnd
+            !$acc loop vector
+            do k = 1,nVertLevels
+               ruAvg_split(k,iEdge) = ruAvg(k,iEdge)
+            end do
+         end do
+         !$acc loop gang worker
+         do iCell = cellStart,cellEnd
+            !$acc loop vector
+            do k = 1,nVertLevels+1
+               wwAvg_split(k,iCell) = wwAvg(k,iCell)
+            end do
+         end do
+         !$acc end parallel
       else
-         ruAvg_split(:,edgeStart:edgeEnd) = ruAvg(:,edgeStart:edgeEnd)+ruAvg_split(:,edgeStart:edgeEnd)
-         wwAvg_split(:,cellStart:cellEnd) = wwAvg(:,cellStart:cellEnd)+wwAvg_split(:,cellStart:cellEnd)
+         !$acc parallel
+         !$acc loop gang worker
+         do iEdge = edgeStart,edgeEnd
+            !$acc loop vector
+            do k = 1,nVertLevels
+               ruAvg_split(k,iEdge) = ruAvg(k,iEdge) + ruAvg_split(k,iEdge)
+            end do
+         end do
+         !$acc loop gang worker
+         do iCell = cellStart,cellEnd
+            !$acc loop vector
+            do k = 1,nVertLevels+1
+               wwAvg_split(k,iCell) = wwAvg(k,iCell) + wwAvg_split(k,iCell)
+            end do
+         end do
+         !$acc end parallel
       end if
 
       if (dynamics_substep == dynamics_split) then
-         ruAvg(:,edgeStart:edgeEnd) = ruAvg_split(:,edgeStart:edgeEnd) * inv_dynamics_split
-         wwAvg(:,cellStart:cellEnd) = wwAvg_split(:,cellStart:cellEnd) * inv_dynamics_split
-         rho_zz_1(:,cellStart:cellEnd) = rho_zz_old_split(:,cellStart:cellEnd)
+         !$acc parallel
+         !$acc loop gang worker
+         do iEdge = edgeStart,edgeEnd
+            !$acc loop vector
+            do k = 1,nVertLevels
+               ruAvg(k,iEdge) = ruAvg_split(k,iEdge) * inv_dynamics_split
+            end do
+         end do
+         !$acc loop gang worker
+         do iCell = cellStart,cellEnd
+            !$acc loop vector
+            do k = 1,nVertLevels+1
+               wwAvg(k,iCell) = wwAvg_split(k,iCell) * inv_dynamics_split
+            end do
+         end do
+         !$acc loop gang worker
+         do iCell = cellStart,cellEnd
+         !$acc loop vector
+         do k = 1,nVertLevels           
+            rho_zz_1(k,iCell) = rho_zz_old_split(k,iCell)
+         end do
+         end do
```

Comment:
It would be good to indent the inner loop to match the others above.

Reply:
Thanks. Done.